android/src/main/java/io/sentry/RNSentryModule.java

```diff
@@ -190,4 +219,114 @@ private Level logLevel(int level) {
                 return Level.OFF;
         }
     }
+
+    @ReactMethod
+    public void setUser(final ReadableMap user) {
+        if (user == null) {
+            Sentry.setUser(null);
+        } else {
+            User userInstance = new User();
+
+            // only support these defined keys for now.
```

Comment:
We'd need to have this clear in our documentation.

This could be a source of support requests where user use our public documented API such as setting `user.data` and not getting it through while the other fields go through correctly

Reply:
Ah, in the javascript SDK I was looking at user at we allow any key, 
```
export interface User {
  [key: string]: any;
  id?: string;
  ip_address?: string;
  email?: string;
  username?: string;
}
```
So I wasn't sure at first how to pass this to native, but I just looked at `User` in Android and it looks like there is a `setOthers` method, I can take every other key that isn't in the defined 4 keys and pass it to others then.